pkgs/development/libraries/libliftoff/default.nix

```
@@ -14,6 +22,21 @@ let
     };
 in
 {
-  libliftoff_0_4 = mkVariant "0.4.1" "sha256-NPwhsd6IOQ0XxNQQNdaaM4kmwoLftokV86WYhoa5csY=";
-  libliftoff_0_5 = mkVariant "0.5.0" "sha256-PcQY8OXPqfn8C30+GAYh0Z916ba5pik8U0fVpZtFb5g=";
+  libliftoff_0_4 = mkVariant {
+    version = "0.4.1";
+    hash = "sha256-NPwhsd6IOQ0XxNQQNdaaM4kmwoLftokV86WYhoa5csY=";
+    patches = [
+      # Pull gcc-14 fix:
+      #   https://gitlab.freedesktop.org/emersion/libliftoff/-/merge_requests/78
+      (fetchpatch {
+        name = "gcc-14-calloc.patch";
```

Nit: It's always nicer to hint which package the patch applies to in the store path.
```suggestion
        name = "libliftoff-gcc-14-calloc.patch";
```